src/components/views/spaces/SpaceTreeLevel.tsx

```diff
@@ -139,6 +139,7 @@ export const SpaceButton = forwardRef<HTMLElement, IButtonProps>(({
             inputRef={handle}
             tabIndex={tabIndex}
             onFocus={onFocus}
+            data-testid="space-button"
```

Comment:
Looks like this is overriding `create-space-button` passed in as props by some callers causing SpacePanel tests to fail.

Reply:
Also, as the [React Testing Library](https://testing-library.com/docs/queries/about/#priority) docs recommend, we should generally prefer more usage-oriented queries like `getByRole` over `getByTestId`, which relies on properties of the UI not visible to the user.